File: from-3201-to-3300/3264-final-array-state-after-k-multiplication-operations-i/solution.go
func getFinalState(nums []int, k int, multiplier int) []int {
    n := len(nums)
    findAndMul := func() {
        minIdx, minVal := 0, nums[0]
        for i := 1; i < n; i ++ {
            if nums[i] < minVal {
                minVal = nums[i]
                minIdx = i
            }
        }
        nums[minIdx] *= multiplier
    }
    for i := 0; i < k; i ++ {
        findAndMul()
    }
    return nums
}

File: from-3201-to-3300/3264-final-array-state-after-k-multiplication-operations-i/solution.py
class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        result = nums[::]
        min_heap = [(num, i) for i, num in enumerate(nums)]
        heapq.heapify(min_heap)

        for _ in range(k):
            n, i = heapq.heappop(min_heap)
            result[i] *= multiplier

            heapq.heappush(min_heap, (result[i], i))

        return result